### PR TITLE
Standardize and improve docstring

### LINKED ISSUE #542: Convert option object docstring
Convert docstrings according to https://stackoverflow.com/a/14820610/12135039

### PR DESCRIPTION
**Pull request by jamestiotio**
**Public API Changes**
The refactoring to `core/Ros.js` might break some applications. However, the proposed changes would make the public API more straightforward in two ways:

- The implementation code is now aligned with the public API as specified in the docstring. I feel that this is important to minimize possible future confusion over the differences between the API described in the docstring versus the actual implementation.
- For the `getNodeDetails` method, the callback function signature now remains the same, whether there is a `failedCallback` function defined or not. Previously, the function signature differs between the two, which might lead to further confusion.

**Description**
This PR standardizes and improves the docstring, along with a few other small things.

More specifically, this PR does the following (in rough order of significance):

- Add parameter and return types to docstrings.
- Add proper indication of optional parameters to the relevant docstrings.
- Rephrase some docstrings to make them clearer.
- Fix some typos in the docstring.
- Add proper capitalization to docstring.
- Method docstrings that start with a verb are standardized to use verbs without the `-s` suffix.

Additionally, there are some non-docstring changes (again, in rough order of significance):

- Slightly refactor `core/Ros.js` to align the callback parameters with the ones specified in the corresponding docstrings. Tests are also updated to follow the new callback function signatures.
- Align the parameter names of the methods `setSucceeded`, `setAborted`, and `sendFeedback` in `actionlib/SimpleActionServer.js` with the respective docstrings.
- Remove dead code in `actionlib/ActionListener.js` (the options `timeout`, `omitFeedback`, `omitStatus`, and `omitResult` are not used there).
- Lint some unnecessary whitespace lines in `actionlib/SimpleActionServer.js` and `core/SocketAdapter.js`.

**Relevant GitHub issues:**
This PR closes #542.

### CODE REVIEW

**Review by Rayman — CHANGES_REQUESTED** [review]
Breaking public API is for me a no-no. It means that every roslib application that has ever been made must now be upgraded.

Standardize and improve docstrings is very welcome. Maybe you could split your PR in 2?

**Comment by MatthijsBurgh** [comment]
@jamestiotio Thanks for your efforts. Like @Rayman said the changes in the documentation is very welcome.

I think any breaking change is not accepted. As I don't plan any major version bump any time soon (and not so soon as well).

**Comment by jamestiotio** [comment]
> Breaking public API is for me a no-no.

Understood. In that case, I have reverted the breaking changes. Essentially, I have aligned the docstrings to simply follow the implementation as the source of truth.

To avoid confusion, I used the `@also` tag for the `getNodeDetails` method in `core/Ros.js` to specify the two different possible callback function signatures. Do let me know if there is a better way to handle this.

**Comment by MatthijsBurgh** [comment]
Please cherry-pick https://github.com/RobotWebTools/roslibjs/commit/605ef08101e47d1cc0c20ca43a5c1d6f2cd2735f in your branch. Than we are good to go.

Thanks for your efforts.

**Comment by jamestiotio** [comment]
@MatthijsBurgh Commit has been cherry-picked.

**Review by MatthijsBurgh — APPROVED** [review]

File: src/core/Ros.js [hunk 1]
```diff
@@ -17,18 +17,18 @@ var EventEmitter2 = require('eventemitter2').EventEmitter2;
...
- *   * groovyCompatibility - don't use interfaces that changed after the last groovy release or rosbridge_suite and related tools (defaults to true)
- *   * transportLibrary (optional) - one of 'websocket', 'workersocket' (default), 'socket.io' or RTCPeerConnection instance controlling how the connection is created in `connect`.
- *   * transportOptions (optional) - the options to use use when creating a connection. Currently only used if `transportLibrary` is RTCPeerConnection.
+ * @param {Object} options
+ * @param {string} [options.url] - The WebSocket URL for rosbridge or the node server URL to connect using socket.io (if socket.io exists in the page). Can be specified later with `connect`.
+ * @param {boolean} [options.groovyCompatibility] - Don't use interfaces that changed after the last groovy release or rosbridge_suite and related tools (defaults to true).
+ * @param {string} [options.transportLibrary] - One of 'websocket', 'workersocket' (default), 'socket.io' or RTCPeerConnection instance controlling how the connection is created in `connect`.
```
Comment (MatthijsBurgh) [inline src/core/Ros.js:30]:
```suggestion
 * @param {string} [options.transportLibrary=websocket] - One of 'websocket', 'workersocket', 'socket.io' or RTCPeerConnection instance controlling how the connection is created in `connect`.
```

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Some changes related optional/default values

**Review by MatthijsBurgh — APPROVED** [review]

**Pull request merged**